File: codesignal/arcade/2.core/8.mirror lake/6.differentSquares.py
# Given a rectangular matrix containing only digits, calculate the number of different 
# 2 × 2 squares in it.

class Solution:
    def differentSquares(self, matrix):
        # Create a set to store the unique squares
        squares = set()
        # Loop through the matrix
        for i in range(len(matrix) - 1):
            for j in range(len(matrix[i]) - 1):
                # Create a square
                square = (matrix[i][j], matrix[i][j + 1], matrix[i + 1][j], matrix[i + 1][j + 1])
                # Add the square to the set
                squares.add(square)
        # Return the length of the set
        return len(squares)

File: codesignal/arcade/2.core/8.mirror lake/6.differentSquares.cpp
// Given a rectangular matrix containing only digits, calculate the number of different 
// 2 × 2 squares in it.

#include <set>
#include <vector>
#include <utility>

class Solution {
public:
  int differentSquares(std::vector<std::vector<int>> matrix) {
    std::set<std::pair<int, int>> r;
    for (int i = 0; i < matrix.size() - 1; ++i)
      for (int j = 0; j < matrix[i].size() - 1; ++j)
        r.insert({matrix[i][j] * 1000 + matrix[i][j + 1] * 100 + matrix[i + 1][j] * 10 + matrix[i + 1][j + 1], 0});
    return r.size();
  }
};